### PR TITLE
feat: upgrade brc-20 api to bestinslot v3, closes #4970

### LINKED ISSUE #4970: Upgrade brc-20 balance API
It seems current v1 bestinslot api is not working and we need to upgrade it to v3 https://docs.bestinslot.xyz/reference/api-reference/ordinals-and-brc-20-and-bitmap-v3-api-mainnet+testnet/brc-20

screenshot_2024-02-22_at_11 11 15_720

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of Leather — [Extension build](https://github.com/leather-wallet/extension/actions/runs/8082587062), [Test report](https://leather-wallet.github.io/playwright-reports/fix/brc20-api-tokens)

This pr upgrades brc-20 tokens queries to bestinslot v3, also adds brc-20 query for Native Segwit address

### CODE REVIEW

**Comment by 314159265359879** [comment]
I don't see the balance loading
image

File: src/app/query/bitcoin/bitcoin-client.ts [hunk 1]
```diff
@@ -51,7 +51,45 @@ export interface BestinslotInscriptionsByTxIdResponse {
...
+  transferrable_balance: string;
+  image_url: string | null;
+}
+
+export interface Brc20Token extends Brc20TokenResponse {
+  decimals: number;
+  holderAddress: string;
+}
+
+interface Brc20TokenTicker {
```
Comment (fbwoolf) [inline src/app/query/bitcoin/bitcoin-client.ts:67]: Should this also have `Response` in the name?

**Comment by alter-eggo** [comment]
@314159265359879 could you please send screenshot with request from network tab

File: src/app/query/bitcoin/bitcoin-client.ts [hunk 2]
```diff
@@ -51,7 +51,45 @@ export interface BestinslotInscriptionsByTxIdResponse {
...
+  ticker: string;
+  overall_balance: string;
+  available_balance: string;
+  transferrable_balance: string;
+  image_url: string | null;
+}
+
+export interface Brc20Token extends Brc20TokenResponse {
+  decimals: number;
+  holderAddress: string;
```
Comment (fbwoolf) [inline src/app/query/bitcoin/bitcoin-client.ts:64]: Do we typically mix response types with camelCased props in client types?

**Review by fbwoolf — APPROVED** [review]

**Comment by 314159265359879** [comment]
> @314159265359879 could you please send screenshot with request from network tab

image

**Comment by 314159265359879** [comment]
> @314159265359879 could you please send screenshot with request from network tab

This is from the correct wallet. The other one is testing a wallet with hardware device. I don't see the calls there. 

image

**Comment by alter-eggo** [comment]
@314159265359879 could you pls send as curl the error request 
Screenshot 2024-02-28 at 19 04 13

**Comment by alter-eggo** [comment]
@314159265359879 it's bc I haven't provided api key, pls check one more time, once build is finished

**Comment by 314159265359879** [comment]
I am in transit now but will check again when back, in ~3 hours

**Review by 314159265359879 — COMMENTED** [review]
LGTM now

**Comment by 314159265359879** [comment]
It doesn't look consistent accross wallets. here I do not see the balance loading. 
I am expecting some BRC20 sats here
image

I do not see console errors either though, this is with hardware device.

File: src/app/query/bitcoin/bitcoin-client.ts (hunk 1 above)
Comment (alter-eggo) [reply src/app/query/bitcoin/bitcoin-client.ts:67]: hm, there is already in the file
```
interface Brc20TickerResponse {
  data: Brc20TokenTicker;
  block_height: number;
}
```

File: src/app/query/bitcoin/bitcoin-client.ts (hunk 2 above)
Comment (alter-eggo) [reply src/app/query/bitcoin/bitcoin-client.ts:64]: don't understand what you mean here :( 
I just need to extend response token type here to use in the app

**Pull request merged**